lib/connection/connection.js

```diff
@@ -205,6 +209,47 @@ var closeHandler = function(self) {
   }
 }
 
+var decompress = function(compressorId, compressedData, callback) {
+  var decompressors = [
```

Comment:
nit: this all looks a little too clever imho. At very least lets build the supported decompressor list outside this method, so we're not building the array every time `decompress` is called?